GridGrinder/Program.cs

```diff
@@ -3,13 +3,14 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
+using Utilities;
 using VRage;
-using VRage.Game.GUI.TextPanel;
 
 // state machine inspired by https://github.com/malware-dev/MDK-SE/wiki/Coroutines---Run-operations-over-multiple-ticks
 
 namespace IngameScript {
 	partial class Program : MyGridProgram {
+		#region mdk preserve

```

Comment:
It's a ... strong hint ... to the MDK minifier to ignore the region. Some minification still happens (eg., comments are still stripped), but the newlines and variable names are maintained (where full minification will try to rename variables to opaque single- or double-character names).

So, this will minify to something like

```
private const string PistonGroupX = "[X]";
private const string PistonGroupY = "[Y]";
private const string PistonGroupZ = "[Z]";
private const float fillMax = 0.9F;

private const string SensorDetectedCommand = "true";
private const string SensorStoppedDetectingCommand = "false";
private const string CalculateGridCargoCommand = "calculate";

private const float blockWidth = 2.5F;
private const float pistonVelocity = 0.5F;
private readonly List<IMyPistonBase> X = new List<IMyPistonBase>();private readonly List<IMyPistonBase> Y = new
List<IMyPistonBase>();private readonly List<IMyPistonBase> Z = new List<IMyPistonBase>();private readonly
List<IMyCargoContainer> a = new List<IMyCargoContainer>();private IMyShipGrinder b;private IMySensorBlock c;
private MyFixedPoint d;
```

Reply:
Ah, i see.